service-lib/src/main/resources/fi/espoo/voltti/logging/logback/default-appender-sanitized.xml

```diff
@@ -38,8 +38,8 @@ Sanitized default appender logback configuration provided for import
           <pattern>
             {
             <!--Hide social security numbers in message and in stack trace-->
-            "message": "%replace(%message){'(\\d{2})(\\d{2})(\\d{2})[A+-](\\d{3})[\\dA-Z]', 'REDACTED-SSN'}",
-            "stackTrace": "%replace(%stacktrace{60,40,32000,rootFirst,inlineHash}){'(\\d{2})(\\d{2})(\\d{2})[A+-](\\d{3})[\\dA-Z]', 'REDACTED-SSN'}",
+            "message": "%replace(%message){'(\\d{2})(\\d{2})(\\d{2})[Aa+-](\\d{3})[\\dA-z](.{0}$|[^-])', 'REDACTED-SSN'}",
+            "stackTrace": "%replace(%stacktrace{60,40,32000,rootFirst,inlineHash}){'(\\d{2})(\\d{2})(\\d{2})[Aa+-](\\d{3})[\\dA-z](.{0}$|[^-])', 'REDACTED-SSN'}",
```

Comment:
Don't yet have a good solution for this but the UUID "e1130765-2925-4549-b395-922222a8208a" would still be match (the last section).

Reply:
Good catch. We could do the same check for the start of the SSN as to the end of SSN: Not treating the SSN as SSN if there is a `-` at the start or at the end of the SSN.

In that case for example these string are not redacted: `-130732-555A`. `130732-555A-`. What do you think?

Reply:
I would be fine with that, there's no perfect solution for this. It could be that the syntax here doesn't support negative lookbehinds but lets hope not.

Reply:
Checking for anything before, even no character at all, but `-` character can be done with `(.{0}^|[^-])` regex so adding it to the beginning of the regex did the trick

Reply:
Is there any functional difference between `.{0}^` and `^`? Quick regex101 check would suggest no (in this case).

Also: not sure how the replacing function works but if it processes the fields as multiline strings, this would eat the newline from a previous line like:

```txt
Some text on first line
000000-0000A
```
would become: `Some text on first lineREDACTED-SSN` if I'm not totally mistaken (regex is hard).

Reply:
I unfortunately don't have the time this week but maybe a brainstorming session would be good; this feels like a brittle direction we are heading towards.

Reply:
Yeah you are right and I'm out of ideas. Let's brainstorm when you have time.